craft_application/application.py

```diff
@@ -94,6 +94,10 @@ class Application:
     :ivar services: A ServiceFactory for this application
     """
 
+    # Feature flag for build-time secrets. Applications that want to use the feature
+    # must set this to True.
+    enable_build_secrets = False
```

Comment:
So since this is a very new, sensitive feature I wanted a way to have the code present & tested but be opt-in by the applications. The idea is that I'll then do a feature-branch in Rockcraft that enables this, while the code that will be in v1.0 won't.
(cc @sergiusens)

Reply:
this is the way!